command/agent.go

```diff
@@ -338,10 +351,43 @@ func (c *AgentCommand) Run(args []string) int {
 
 	ctx := context.Background()
 
+	// Generate the Nomad client.
+	if err := c.agent.GenerateNomadClient(); err != nil {
+		logger.Error("failed to start the nomad client", "error", err)
+		return 1
+	}
+
+	if parsedConfig.HighAvailability.Enable {
```

Comment:
Just for readability, I wonder if we could use a switch statement to control this?
```
	switch parsedConfig.HighAvailability.Enable {
	case true:
	default:
	}
```